pkg/instrumentation/annotation.go

```diff
@@ -27,6 +27,7 @@ const (
 	annotationInjectNodeJS        = "instrumentation.opentelemetry.io/inject-nodejs"
 	annotationInjectPython        = "instrumentation.opentelemetry.io/inject-python"
 	annotationInjectDotNet        = "instrumentation.opentelemetry.io/inject-dotnet"
+	annotationDotNetRuntime       = "instrumentation.opentelemetry.io/otel-dotnet-auto-runtime"
```

Comment:
Annotation name is different than discussed in #1849
I have tried to make it compatible with `nnotationGoExecPath`